programs/disks/disks.cpp

```diff
@@ -0,0 +1,170 @@
+{
+    if (options.count("config-file"))
+        config().setString("config-file", options["config-file"].as<String>());
+    if (options.count("disk"))
+        config().setString("disk", options["disk"].as<String>());
+    if (options.count("diskFrom"))
+        config().setString("diskFrom", options["diskFrom"].as<String>());
+    if (options.count("diskTo"))
+        config().setString("diskTo", options["diskTo"].as<String>());
+    if (options.count("input"))
+        config().setString("input", options["input"].as<String>());
+    if (options.count("output"))
+        config().setString("output", options["output"].as<String>());
+}
+
+void DisksApp::init(std::vector<String> & common_arguments)
+{
+    stopOptionsProcessing();
+
+    po::variables_map options;
+    std::optional<ProgramOptionsDescription> options_description;
+    options_description.emplace(createOptionsDescription("clickhouse-disks", getTerminalWidth()));
+
+    po::positional_options_description positional_options_description;
+
+    addOptions(options_description, positional_options_description);
+
+    parseAndCheckOptions(options_description, positional_options_description, options, common_arguments);
+
+    po::notify(options);
+
+    if (options.count("help"))
+    {
+        command_descriptions["clickhouse-disks"]->printHelpMessage();
+        exit(0);
+    }
+
+    if (command_arguments.size() == 0 || !supported_commands.count(command_arguments[0]))
+    {
+        command_descriptions["clickhouse-disks"]->printHelpMessage();
+        throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+    }
+
+    processOptions(options_description, options);
+
+    argsToConfig(common_arguments, config(), 100);
+}
+
+void DisksApp::parseAndCheckOptions(std::optional<ProgramOptionsDescription> & options_description,
+                                    boost::program_options::positional_options_description & positional_options_description,
+                                    po::variables_map & options,
+                                    Arguments & arguments)
+{
+    auto parser = po::command_line_parser(arguments).options(options_description.value()).positional(positional_options_description).allow_unregistered();
+    po::parsed_options parsed = parser.run();
+    po::store(parsed, options);
+}
+
+int DisksApp::main(const std::vector<String> & /*args*/)
+{
+    String path = config().getString("path", DBMS_DEFAULT_PATH);
+
+    loadConfiguration();
+
+    registerDisks();
+    registerFormats();
+
+    shared_context = Context::createShared();
+    global_context = Context::createGlobal(shared_context.get());
+
+    global_context->makeGlobalContext();
+    global_context->setApplicationType(Context::ApplicationType::DISKS);
+    global_context->setPath(path);
+
+    command_descriptions[command_arguments[0]]->execute(command_arguments, global_context, config());
```

Comment:
let's save `command_arguments[0]` into a separate variable `command_name` in class when validating command arguments . And same separate variable for `command_arguments`, so that they do not contain command name, only command arguments.